packages/components/src/tooltip/test/index.js

```diff
@@ -23,171 +28,136 @@ describe( 'Tooltip', () => {
-				} )
+			await waitFor( () =>
+				expect( screen.getByText( 'tooltip text' ) ).toBeVisible()
 			);
 
 			// Wait for the tooltip element to be positioned (aligned with the button)
 			await waitFor( () =>
 				expect(
-					getWrappingPopoverElement( tooltip )
+					getWrappingPopoverElement(
+						screen.getByText( 'tooltip text' )
+					)
 				).toBePositionedPopover()
 			);
 		} );
 
-		it( 'should render children with additional tooltip when hovered', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
+		it( 'should render the tooltip when anchor is hovered', async () => {
+			const user = userEvent.setup();
 
 			render(
-				<Tooltip text="Help text">
-					<button>Hover Me!</button>
+				<Tooltip { ...props }>
+					<Button>Hover Me!</Button>
 				</Tooltip>
 			);
 
 			const button = screen.getByRole( 'button', { name: 'Hover Me!' } );
-			expect( button ).toBeVisible();
 
 			await user.hover( button );
 
-			// Tooltip hasn't appeared yet
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-
-			act( () => jest.advanceTimersByTime( TOOLTIP_DELAY ) );
-
-			// Tooltip shows after the delay
-			const tooltip = screen.getByText( 'Help text' );
-			expect( tooltip ).toBeVisible();
+			await waitFor( () =>
+				expect( screen.getByText( 'tooltip text' ) ).toBeVisible()
+			);
 
 			// Wait for the tooltip element to be positioned (aligned with the button)
 			await waitFor( () =>
 				expect(
-					getWrappingPopoverElement( tooltip )
+					getWrappingPopoverElement(
+						screen.getByText( 'tooltip text' )
+					)
 				).toBePositionedPopover()
 			);
+
+			await user.unhover( button );
+
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
 		} );
 
 		it( 'should not show tooltip on focus as result of mouse click', async () => {
-			const user = userEvent.setup( {
-				advanceTimers: jest.advanceTimersByTime,
-			} );
-			const mockOnFocus = jest.fn();
+			const user = userEvent.setup();
 
 			render(
-				<Tooltip text="Help text">
-					<button onFocus={ mockOnFocus }>Hover Me!</button>
+				<Tooltip { ...props }>
+					<Button>Hover Me!</Button>
 				</Tooltip>
 			);
 
-			const button = screen.getByRole( 'button', { text: 'Hover Me!' } );
-			expect( button ).toBeVisible();
-
-			await user.click( button );
-
-			// Tooltip hasn't appeared yet
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-
-			act( () => jest.advanceTimersByTime( TOOLTIP_DELAY ) );
-
-			// Tooltip still hasn't appeared yet, even though the component was focused
-			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-			expect( mockOnFocus ).toHaveBeenCalledWith(
-				expect.objectContaining( {
-					type: 'focus',
-				} )
+			await user.click(
+				screen.getByRole( 'button', { text: 'Hover Me!' } )
 			);
+
+			expect(
+				screen.queryByText( 'tooltip text' )
+			).not.toBeInTheDocument();
```

Comment:
How are we certain that the tooltip won't appear a bit after we ran this assertion? The previous version of the test was waiting for some time and doing the check again.

Reply:
If a button is clicked and the cursor doesn't move, then the tooltip would appear after reaching the set `delay`. But if there was no `delay`, a mouse click would trigger it. My understanding is this test is assuring that an initial mouse click triggers the focus without a tooltip if there is a delay set. 

If that's not the intention, then would it make sense to click and move the cursor off of the button? Or to combine this with test `should not show tooltip if the mouse leaves the anchor before the tooltip has shown` and have the first interaction be a mouse click instead of hover. Thoughts?

Reply:
I think we're talking about different things.

Previously, the test was waiting for some time before checking if the tooltip was there or not. Now, we're not waiting for any time - we just perform the assertion after the click. So because we didn't wait enough, we can't be sure that after the assertion, the tooltip won't appear. 

Also, I get your point about the `delay` but I don't see us setting a delay for this test, and AFAIK it's `0` by default? 

Let me know if I still don't make sense.

Reply:
By default, the delay is set to 700ms. I don't think there are many cases where we'd want a delay of 0 for a tooltip because then it would show up on a mouse click or a swift movement across the element 😄 

Would you find it preferable if we added a timeout to the assertion? i.e.

```js
await waitFor(
	() =>
		expect(
			screen.queryByText( 'tooltip text' )
		).not.toBeInTheDocument(),
	{ timeout: TOOLTIP_DELAY }
);
```

Reply:
I might be missing something, but isn't the `delay` set to `0` by default? That's what the documentation indicates too:

https://testing-library.com/docs/user-event/options/#delay

Unless we alter it when we're calling `userEvent.setup()`, which we don't seem to be doing in this test.

Reply:
Ah, we are talking about two different delays! I'm referring to the `Tooltip` prop and now I see you are referring to `userEvent` options. 🙂 

I'm curious why adding a timeout wouldn't have the same effect?

Reply:
I'm not saying it won't, was just trying to clarify. To resolve the primary topic here: I think some additional timeout in a `waitFor()` as you suggested will do the job as well 👍